plugins/common/kafka/config.go

```diff
@@ -102,5 +124,27 @@ func (k *Config) SetConfig(config *sarama.Config) error {
 		config.Metadata.Full = *k.MetadataFull
 	}
 
+	if k.MetadataRetryMax != 0 {
+		config.Metadata.Retry.Max = k.MetadataRetryMax
+	}
+
+	if k.MetadataRetryBackoff != 0 {
+		// If config.Metadata.Retry.BackoffFunc is set, sarama ignores
+		// config.Metadata.Retry.Backoff
+		config.Metadata.Retry.Backoff = k.MetadataRetryBackoff
+	}
+
+	switch strings.ToLower(k.MetadataRetryType) {
+	default:
+		return fmt.Errorf("invalid metadata retry type")
+	case "exponential":
+		if k.MetadataRetryBackoff != 0 {
+			config.Metadata.Retry.BackoffFunc = makeBackoffFunc(k.MetadataRetryBackoff, k.MetadataRetryMaxDuration)
+		}
+	case "constant":
+		fallthrough
+	case "":
```

Comment:
Should we catch `k.MetadataRetryBackoff == 0` and `k.MetadataRetryType == "constant"` and output an error as this would lead to no backoff if my understanding is correct!?

Reply:
Yes, that is a good idea.

Reply:
If retry type is constant (or unset) and the backoff is unset (zero), it uses the sarama default of 250 ms, like it did before this PR. I made note of this in the sample config.

Looking closer at this code though, if retry type is exponential and backoff is zero, it won't use exponential retry and it won't log a warning. I'll add a default backoff and log a warning.

Reply:
Added a default backoff so this should be done